### ISSUE TITLE
Missing property 'access-key' when 'iam-role' is specified.

### ISSUE THREAD

**Issue by acisternino** [hazelcast/hazelcast-aws#32]
Hi.

I am trying to deploy Hazelcast on EC2 using IAM roles for discovering other cluster members and I am getting the following error:
```
Exception in thread "main" java.lang.RuntimeException: Failed to configure discovery strategies
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.loadDiscoveryStrategies(DefaultDiscoveryService.java:168)
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.<init>(DefaultDiscoveryService.java:60)
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryServiceProvider.newDiscoveryService(DefaultDiscoveryServiceProvider.java:29)
	at com.hazelcast.instance.Node.createDiscoveryService(Node.java:259)
	at com.hazelcast.instance.Node.<init>(Node.java:213)
	at com.hazelcast.instance.HazelcastInstanceImpl.createNode(HazelcastInstanceImpl.java:159)
	at com.hazelcast.instance.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:127)
	at com.hazelcast.instance.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:218)
	at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:176)
	at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:126)
	at com.hazelcast.core.Hazelcast.newHazelcastInstance(Hazelcast.java:58)
	at com.hazelcast.core.server.StartServer.main(StartServer.java:46)
Caused by: com.hazelcast.core.HazelcastException: Missing property 'access-key' on discovery strategy 'com.hazelcast.aws.AwsDiscoveryStrategy' configuration
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.buildProperties(DefaultDiscoveryService.java:188)
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.buildDiscoveryStrategy(DefaultDiscoveryService.java:216)
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.loadDiscoveryStrategies(DefaultDiscoveryService.java:158)
	... 11 more
```
This is my hazelcast.xml file:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<hazelcast xsi:schemaLocation="http://www.hazelcast.com/schema/config hazelcast-config-3.8.xsd"
           xmlns="http://www.hazelcast.com/schema/config"
           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">

    <!--
     | Hazelcast configuration file for running a default-configured cluster
     | on AWS.
    -->
    <properties>
        <property name="hazelcast.discovery.enabled">true</property>
    </properties>

    <group>
        <name>dev</name>
        <password>dev-pass</password>
    </group>

    <management-center enabled="false">http://localhost:8080/mancenter</management-center>

    <network>
        <port auto-increment="true" port-count="50">5701</port>
        <join>
            <multicast enabled="false"/>
            <tcp-ip enabled="false"/>
            <aws enabled="false"/>
            <discovery-strategies>
                <!-- See: https://github.com/hazelcast/hazelcast-aws -->
                <!-- "class" attribute equals to the DiscoveryStrategy not the factory! -->
                <discovery-strategy enabled="true" class="com.hazelcast.aws.AwsDiscoveryStrategy">
                    <properties>
                       <property name="iam-role">PippoRole</property>
                       <property name="region">eu-central-1</property>
                       <property name="tag-key">role</property>
                       <property name="tag-value">hazelcast-node</property>
                       <property name="connection-timeout-seconds">20</property>
                    </properties>
                </discovery-strategy>
            </discovery-strategies>
        </join>
    </network>

    <partition-group enabled="false"/>

    <executor-service name="default">
        <pool-size>16</pool-size>
        <!-- Queue capacity. 0 means Integer.MAX_VALUE. -->
        <queue-capacity>0</queue-capacity>
    </executor-service>
</hazelcast>
```
`PippoRole` has the following policy attached:
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "ec2:DescribeInstances"
            ],
            "Resource": [
                "*"
            ]
        }
    ]
}
```
and it is correctly attached to the running instance.

From inside the instance, when I try to get the auth token from the metadata, I get
```
$ curl http://169.254.169.254/latest/meta-data/iam/security-credentials/PippoRole
{
  "Code" : "Success",
  "LastUpdated" : "2017-07-24T13:14:53Z",
  "Type" : "AWS-HMAC",
  "AccessKeyId" : " ... key text ... ",
  "SecretAccessKey" : " ... key text ... ",
  "Token" : " ... long token text ... ",
  "Expiration" : "2017-07-24T19:18:56Z"
```
which seems to show that everything is OK.

* Hazelcast is v3.8.3
* Hazelcast AWS is v2.0.1
* OS is Ubuntu Linux Xenial fully updated

What am I doing wrong?

**Comment by mesutcelik** [comment]
Hi @acisternino ,
Can you please try to use 2.0.2-SNAPSHOT?

It must be fixed by this PR. https://github.com/hazelcast/hazelcast-aws/pull/30

**Comment by acisternino** [comment]
Thanks for the quick reply @mesutcelik.

I'll try in a couple of hours but the PR looks like the right one.

**Comment by acisternino** [comment]
First tests looks good.

Server instances are discovered.  👍

**Comment by emrahkocaman** [comment]
Closing this issue as it's duplicate of https://github.com/hazelcast/hazelcast-aws/issues/29 and already resolved and addressed for 2.0.2.
Thanks for reporting @acisternino
